### PR TITLE
Adding FIPS 140-3 Windows build support

### PR DESCRIPTION
**Pull request by tim-weller-wolfssl**
Tested in Windows 11 environment using Visual Studio 2022.

### CODE REVIEW

File: windows/fips_140_3/user_settings.h [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+#define WE_HAVE_ECKEYGEN
+#define WE_HAVE_EC_P192
+#define WE_HAVE_EC_P224
+#define WE_HAVE_EC_P256
+#define WE_HAVE_EC_P384
+#define WE_HAVE_EC_P521
+#define WE_HAVE_DIGEST
+
+/* wolfSSL's FIPS 140-3 cert doesn't have DES3 in the module. */
+/* #define WE_HAVE_DES3CBC */
```
Comment (tim-weller-wolfssl) [inline windows/fips_140_3/user_settings.h:40]: Duplicate line/comment.  This instance at line 40 will be removed.

**Comment by JacobBarthelmeh** [comment]
- Had to add ws2_32.lib to the wolfEngine "test" projects linker->Input for the socket API
- The 'Debug Fips 140-3' x86 build configuration was looking for the wolfSSL library to be in wolfssl/IDE/WIN10/DLL Debug/ but when I was building wolfSSL as Win32 and debug it was located in wolfssl/IDE/WIN10/Debug/
- Post copy command ran into an error

```
Severity	Code	Description	Project	File	Line	Suppression State
Error	MSB3073	The command "copy /Y "D:\work\wolfengine\windows\..\..\openssl\*.dll" "D:\work\wolfengine\windows\Debug FIPS 140-3\Win32\"
copy /Y "D:\work\wolfengine\windows\..\..\wolfssl\IDE\WIN10\DLL Debug\Win32\*.dll" "D:\work\wolfengine\windows\Debug FIPS 140-3\Win32\"
:VCEnd" exited with code 1.	test	C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Microsoft\VC\v170\Microsoft.CppCommon.targets	157	
```

The files do seem to exist

```
PS D:\work\openssl> ls *.dll


    Directory: D:\work\openssl


Mode                 LastWriteTime         Length Name
----                 -------------         ------ ----
-a----         9/15/2022   2:34 PM        2255872 libcrypto-1_1.dll
-a----         9/15/2022   2:34 PM         538624 libssl-1_1.dll
```

**Comment by tim-weller-wolfssl** [comment]
@JacobBarthelmeh, thank-you for the feedback!  I have missed some testing permutations.  Let me revisit testing, and update the PR accordingly.

**Comment by haydenroche5** [comment]
@tim-weller-wolfssl you can assign me back to this when you're ready for another round of review.

**Comment by haydenroche5** [comment]
@tim-weller-wolfssl It looks like your latest push has a merge commit in it. Please rebase your local commits onto upstream master and push that way. Your commits should be linear on top of master. Ping me if you need further assistance.

**Comment by tim-weller-wolfssl** [comment]
It was determined that initially, at least, FIPS 140-3 on Windows will only be buildable as a DLL.  Testing of the added support involved building and running the wolfEngine test in the following configurations in Windows (11), for both *x64* and *Win32* (*x86*):
* OpenSSL
  * with wolfSSL *Debug* and *DLL Debug* builds
    * with wolfEngine *DLL Debug FIPS 140-3* build
  * with wolfSSL *Release* and *DLL Release* builds
    * with wolfEngine *DLL Release FIPS 140-3*

**Review by haydenroche5 — APPROVED** [review]
Tested the new configs locally. Works great!

**Pull request merged**